transcom/mymove
# MB-7359 Services counselor can edit order details

## Description

> As a services counselor, I want to be able to edit the order details for a move, so I can make any necessary adjustments in case of error.  
> 
> **Acceptance criteria:**
> 
> Services counselor can see and edit the following fields on the View orders page:
> 
> - Current duty station
> 
> - New duty station
> 
> - Date issued
> 
> - Report by date
> 
> - Orders type
> 
> Labels and content are consistent with TOO experience.
> 
> Counselor CANNOT see or edit department indicator, orders number, or orders detail type

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?
- This PR separates the endpoints for each order and allowance form
- Did a bunch of refactor on the endpoints and service object that updates the order with entitlements
- Since the UpdateAllowance is basically an order with entitlements, I decided to return an order model which made the most sense to me
- Services Counselor can't edit fields they are not authorized.
- A gotcha about the unauthorized fields for Services Counselor, the fields just doesn't update instead of returning an error. I thought that would be cleaner but let me know if that seems weird

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

Test
1. Log in as a Services Counselor
2. Click on any move that `Needs Counseling`
3. Click on `View and edit order` button
4. Change values of the form
5. Click `Save`

Storybook
```sh
make storybook
```

1. Navigate to `http://localhost:6006/?path=/story/office-components-ordersdetailform--fields-hidden`

You can see the orders detail form hides the fields. Play around with the controls to show/hide fields.

## Code Review Verification Steps

(https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7359) for this change

## Screenshots


https://user-images.githubusercontent.com/1522549/118215973-771c4e00-b427-11eb-8b06-1f1bd8602de2.mov
